texastribune/donations
# Give members access to new ump tools

#### What's this PR do?
Opens up the newest ump tools to members. (ability to edit cc info on ANY/ALL recurring donations, ability to end recurring donation).

#### Why are we doing this? How does it help us?
Gives members greater control over their own membership.

#### How should this be manually tested?
note: a few of these screenshots are somewhat outdated because of updates, but the general functionality stands

* Log in to '/account' locally or on our staging site as a non-staff member
    * <img width="486" alt="Screen Shot 2023-09-14 at 12 49 02 PM" src="https://github.com/texastribune/donations/assets/88053677/4a4373e6-5711-443d-9d13-c7d66a270705">
* Navigate to the Membership tab
    * <img width="484" alt="Screen Shot 2023-09-12 at 9 15 55 AM" src="https://github.com/texastribune/donations/assets/88053677/57594d55-ff8a-4796-b4de-6b6effea87d8">
* You should see multiple recurring donations
   * <img width="1016" alt="Screen Shot 2023-09-14 at 12 51 44 PM" src="https://github.com/texastribune/donations/assets/88053677/2da07af6-a087-4ccb-bf0d-6bd590d31090">
* Choose a recurring donation to edit and fill out the form in the pop-up with a [Stripe test card](https://stripe.com/docs/testing#cards)
    * <img width="809" alt="Screen Shot 2023-09-12 at 9 19 01 AM" src="https://github.com/texastribune/donations/assets/88053677/7d7ebc8d-b68f-472f-9493-6341dbadefa1">
* After the update goes through, you should see a success message with helpful information
    * <img width="521" alt="Screen Shot 2023-09-12 at 9 20 16 AM" src="https://github.com/texastribune/donations/assets/88053677/9e9f2199-5dae-4347-8f3b-9e8a06a14a08">
* Attempt deleting the same recurring donation by clicking the delete button and confirming in the pop-up
    * <img width="598" alt="Screen Shot 2023-09-12 at 9 21 32 AM" src="https://github.com/texastribune/donations/assets/88053677/14deb510-5c16-41c8-a95e-91789aba76f8">
* After the cancellation goes through, you should see a success message with helpful information and a link out to a form
    * <img width="610" alt="Screen Shot 2023-09-12 at 9 22 50 AM" src="https://github.com/texastribune/donations/assets/88053677/b5de9273-91b8-43d2-8658-d4d5f060fbfc">
* Follow the link and make sure it takes you to an airtable form
    * <img width="1080" alt="Screen Shot 2023-09-12 at 9 23 36 AM" src="https://github.com/texastribune/donations/assets/88053677/938b1e55-e8ee-4b8b-9eeb-e01ce57d8e46">

#### How should this change be communicated to end users?
We'll let the membership team know, but we're considering this a soft launch and won't be directly reaching out to members about the updates, as of now.

#### Are there any smells or added technical debt to note?
Plenty of cleaning up happening here, so worth thorough testing.

At some point, it might be worth it to make our gtm constants setup slightly simpler, but I utilized the existing setup for consistency's sake and because overhauling that felt like it was out of scope here.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recQzF8OzVPj8xZLr?blocks=hide
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recMG40yNTJjhn97w?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:
